bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -0,0 +1,832 @@
+            password = locConfig.password;
+            username = locConfig.username;
+
+            // build base URL
+            baseUrl = "http://" + ipAddress + "/fhapi/v1/api";
+
+            // Open Http connection
+            if (!openHttpConnection()) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/comm-error.http-wrongpass-or-ip");
+
+                thingReachable = false;
+            }
+
+            // Open the websocket connection for immediate status updates
+            if (!openWebSocketConnection()) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/comm-error.not-able-open-websocketconnection");
+
+                thingReachable = false;
+            }
+
+            if (thingReachable) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        });
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        super.handleConfigurationUpdate(configurationParameters);
+
+        // load configuration
+        FreeAtHomeBridgeHandlerConfiguration locConfig = getConfigAs(FreeAtHomeBridgeHandlerConfiguration.class);
+
+        ipAddress = locConfig.ipAddress;
+        password = locConfig.password;
+        username = locConfig.username;
+    }
+
+    /**
+     * Method to dispose
+     */
+    @Override
+    public void dispose() {
+        // let run out the thread
+        socketMonitor.interrupt();
+    }
+
+    /**
+     * Thread that maintains connection via Websocket.
+     */
+    private class FreeAtHomeWebsocketMonitorThread extends Thread {
+
+        // initial delay to initiate connection
+        private AtomicInteger reconnectDelay = new AtomicInteger();
+
+        public FreeAtHomeWebsocketMonitorThread() {
+        }
+
+        @Override
+        public void run() {
+            // set initial connect delay to 0
+            reconnectDelay.set(0);
+
+            try {
+                while (!isInterrupted()) {
+                    if (httpConnectionOK.get()) {
+                        if (connectSession()) {
+                            while (!socket.isSocketClosed()) {
+                                TimeUnit.SECONDS.sleep(BRIDGE_WEBSOCKET_KEEPALIVE);
+
+                                logger.debug("Sending keep-alive message {}", System.currentTimeMillis());
+                                socket.sendKeepAliveMessage("keep-alive");
+                            }
+
+                            logger.debug("Socket connection closed");
+                            reconnectDelay.set(BRIDGE_WEBSOCKET_RECONNECT_DELAY);
+                        }
+                    } else {
+                        TimeUnit.SECONDS.sleep(BRIDGE_WEBSOCKET_RECONNECT_DELAY);
+                    }
+                }
+            } catch (InterruptedException e) {
```

Comment:
Restore the interrupted state of the thread by calling `Thread.currentThread().interrupt()`, see https://rules.sonarsource.com/java/RSPEC-2142/